copier/vcs.py

```diff
@@ -137,7 +137,11 @@ def clone(url: str, ref: OptStr = None) -> str:
     location = mkdtemp(prefix=f"{__name__}.clone.")
     _clone = git["clone", "--no-checkout", url, location]
     # Faster clones if possible
-    if GIT_VERSION >= Version("2.27"):
+    if (
+        GIT_VERSION >= Version("2.27")
+        and not url.startswith("file://")
+        and not Path(url).exists()
+    ):
```

Comment:
What kind of problems were caused by local blobless clones?

Reply:
I use Copier to test a custom template in CI and the blobless clones make `git clone` to use like 10GB of memory, which is quite annoying for a simple CI job (it uses about 1GB with this branch). This CI runs in a docker container on a local cluster, so the setup is a bit complex, and I don't really know why git does not like blobless repos in this case... But fetching blobs for local repos should not cause perf issues as local transfers are usually very fast in this case.

Reply:
That's weird because local cloning operations are common across all our test codebase. 

Usually `/tmp` is mounted as `tmpfs`, which means that its files exist in memory. Maybe your repo is too big? Could you try setting a different tmp directory? Or mounting another kind of filesystem there?

Reply:
I performed a few more tests and I think I got it. The real issue is when copier tries to clone a shallow clone. This happens in CI because, by default, Gitlab creates a shallow clone of the tested repository and then I run Copier on this cloned repository.
To reproduce you can try the following commands:
```shell
rm -rf /tmp/test_copier;
mkdir /tmp/test_copier;
cd /tmp/test_copier;

git clone --depth=2 https://github.com/copier-org/autopretty.git autopretty;

git clone --no-checkout autopretty /tmp/test_copier/copy_1 --filter=blob:none;
```
As you will see, many warnings are reported and the process is much longer than expected. It also takes much more memory, though it's harder to monitor.
If you run the same script removing the `--filter=blob:none`, you will see that the second clone does not report any warning and is much faster.

Reply:
I proposed a new patch to just check if the repository is a shallow clone or not.

Reply:
Any comment on this @yajo ?

Reply:
Yeah I see your point.
Nevertheless, I think it would be nice to at least raise a warning in such cases because:
* it can still work (if the shallow clone contains all the required commits).
* even if it works it can lead to unusually high resource consumption (that was my initial issue, which was quite hard to understand).
* if it fails it can be hard to debug.

Reply:
(the warning would be in addition to the doc ofc)

Reply:
Good idea! Wanna update this PR with those changes?

Reply:
Sure, I pushed new commits with a warning and a new FAQ entry